iothub/service/API design.md

```diff
@@ -0,0 +1,88 @@
+
+	protected IotHubServiceClient();
+	
+	public IotHubServiceClient(string connectionString, IotHubServiceClientOptions options = default);
+	public IotHubServiceClient(string hostName, TokenCredential credential, IotHubServiceClientOptions options = default);
+	public IotHubServiceClient(string hostName, AzureSasCredential credential, IotHubServiceClientOptions options = default);
+
+	public void Dispose();
+}
+
+public class IotHubServiceClientOptions
+{
+	public IotHubServiceClientOptions(ServiceVersion version = LatestVersion)
+
+	public IWebProxy Proxy { get; set; }
+	public TimeSpan HttpConnectionLeaseTimeout { get; set; } = TimeSpan.FromMinutes(5);
+	public HttpClient HttpClient { get; set; }
+	public ServiceVersion Version { get; set; } = LatestVersion;
+
+	public enum ServiceVersion
+	{
+		V2021_04_12 = 1,
+		V2020_03_13 = 2,
+		V2019_10_01 = 3,
+		V2019_09_30 = 4,
+		V2019_03_30 = 5,
+		V2018_06_30 = 6,
+		V2018_04_01 = 7
+	}
+}
+
+
+public class DevicesClient
+{
+	protected DevicesClient();
+
+	public virtual async Task<Device> AddAsync(Device device, CancellationToken cancellationToken = default);
+	public virtual async Task<Device> GetAsync(string deviceId, CancellationToken cancellationToken = default);
+	public virtual async Task<Device> SetAsync(Device device, CancellationToken cancellationToken = default);
+	public virtual async Task<Device> SetAsync(Device device, bool forceUpdate, CancellationToken cancellationToken = default);
+	public virtual async Task DeleteAsync(string deviceId, CancellationToken cancellationToken = default);
+	public virtual async Task DeleteAsync(Device device, CancellationToken cancellationToken = default);
+
+	public virtual async Task<BulkRegistryOperationResult> AddWithTwinAsync(Device device, Twin twin, CancellationToken cancellationToken = default);
+
+	public virtual async Task<BulkRegistryOperationResult> AddAsync(IEnumerable<Device> devices, CancellationToken cancellationToken = default);
+	public virtual async Task<BulkRegistryOperationResult> SetAsync(IEnumerable<Device> devices, CancellationToken cancellationToken = default);
+	public virtual async Task<BulkRegistryOperationResult> SetAsync(IEnumerable<Device> devices, bool forceUpdate, CancellationToken cancellationToken = default);
+	public virtual async Task<BulkRegistryOperationResult> DeleteAsync(IEnumerable<Device> devices, CancellationToken cancellationToken = default);
+	public virtual async Task<BulkRegistryOperationResult> DeleteAsync(IEnumerable<Device> devices, bool forceDelete, CancellationToken cancellationToken = default);
+
+	public virtual async Task<IEnumerable<Module>> GetModulesAsync(string deviceId, CancellationToken cancellationToken = default);
+			
+	public virtual async Task<RegistryStatistics> GetRegistryStatisticsAsync(CancellationToken cancellationToken = default);
+	public virtual async Task<ServiceStatistics> GetServiceStatisticsAsync(CancellationToken cancellationToken = default);
```

Comment:
"Service statistics" just includes connected device count currently, so it can fit in this client